stdlib/email/src/main/ballerina/src/email/imap_client_endpoint.bal

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# Represents an IMAP Client that interacts with an IMAP Server
+public type ImapClient client object {
+
+    # Gets invoked during object initialization
+    #
+    # + clientConfig - Configurations for IMAP Client
+    public function __init(ImapConfig clientConfig) {
+        error? response = initImapClientEndpoint(self, clientConfig, false);
+    }
+
+    # Used to read a message
+    #
+    # + return - An `error` if failed to send the message to the recipient
+    public remote function read() returns Email|error? {
+        Filter filter = {
+            folder: "INBOX"
+        };
```

Comment:
Shall we get the filter value as user input without hardcoding to "INBOX"?

Reply:
In this phase just use the hard coded parameter as there are no vital scenarios found other than INBOX.

Reply:
The problem is if we want to add another filter parameter later, we need to change the API. It's ok if we only support INDOX for now. but we need to design the API to incorporate future changes without breaking. So we pass filter as a parameter and through an unsupported error will be a better approach. WDYT?